### ISSUE TITLE
How you use ADetailer with colab without GUI

### ISSUE THREAD

**Issue by Artkoch** [Bing-su/adetailer#430]
### Question

As I'm new to SD and I don't know what I'm doing, please, could someone give an example of ADetailer implementation into a typical SD pipeline with diffusers library in colab or .py file
 
Because I would like to increase details in generated interior designs.

My current code is basic tutorial code

`!pip install opencv-contrib-python
!pip install diffusers transformers accelerate

import torch
import os
from huggingface_hub import HfApi
from pathlib import Path
from diffusers.utils import load_image
import numpy as np
import cv2
from PIL import Image

from diffusers import (
    ControlNetModel,
    StableDiffusionControlNetPipeline,
    UniPCMultistepScheduler,
)

checkpoint = "lllyasviel/control_v11p_sd15_canny"

image = load_image(
    "https://cdn.pixabay.com/photo/2014/08/11/21/39/wall-416060_1280.jpg"
)

image = np.array(image)

low_threshold = 100
high_threshold = 200

image = cv2.Canny(image, low_threshold, high_threshold)
image = image[:, :, None]
image = np.concatenate([image, image, image], axis=2)
control_image = Image.fromarray(image)

control_image.save("./images/control.png")

controlnet = ControlNetModel.from_pretrained(checkpoint, torch_dtype=torch.float16)
pipe = StableDiffusionControlNetPipeline.from_pretrained(
    "runwayml/stable-diffusion-v1-5", controlnet=controlnet, torch_dtype=torch.float16
)

pipe.scheduler = UniPCMultistepScheduler.from_config(pipe.scheduler.config)
pipe.enable_model_cpu_offload()

generator = torch.manual_seed(33)
image = pipe("Korea style room", num_inference_steps=20, generator=generator, image=control_image).images[0]

image.save('images/image_out.png')
image
`

**Comment by khawar-islam** [comment]
Hello @Artkoch did you find the solution? I am also looking for this
